docs/scripts/gen_reference.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3.8
+
+from __future__ import annotations
+import os
+
+import yaml
+import argparse
+
+from pathlib import Path
+from dataclasses import dataclass, asdict
+from typing import Dict, List, Optional, Union
+
+
+@dataclass(frozen=True)
+class ResourceNames:
+    kind: str
+    listKind: str
+    plural: str
+    singular: str
+
+@dataclass(frozen=True)
+class Field:
+    name: str
+    description: str
+    type: str
+    required: bool
+    contains: Optional[Union[str, List[Field]]]
+    contains_description: Optional[str]
+
+@dataclass(frozen=True)
+class Resource:
+    name: str
+    service: str
+    group: str
+    apiVersion: str
+    description: str
+    scope: str
+    names: ResourceNames
+    spec: List[Field]
+    status: List[Field]
+
+    def output_path(self) -> Path:
+        return Path(f"{self.service}/{self.apiVersion}/{self.name}.md")
```

Comment:
might consider lowercasing the `{self.name}`...

Reply:
The `self.name` is the only part of the CRD that is properly cased - and that means it shows up properly cased in the sidebar. Usually I'd agree with the filename being lowercased, but in this case mkdocs is using it for formatting so I think it's preferred.

Reply:
k, makes sense